### ISSUE TITLE
Incorrect formatting for elvis operator with comments over options

### ISSUE THREAD

**Issue by boris-faniuk-n26** [pinterest/ktlint#2153]
## Expected Behavior
This snippet represents good formatting according to IntelliJ IDEA and should survive after ktlint 
```kotlin
    fun get(): String =
        // first way to get value
        first() ?:
        // otherwise use second way
        second()
```

## Observed Behavior
This formatting is applied by ktlint
```kotlin
    fun get(): String =
        // first way to get value
        first()
            ?: // otherwise use second way
            second()
```
This has 2 problems
1.  Second option has indentation despite it has the same meaning as first option
2. Elvis operator is merged into next comment  

## Your Environment

* Version of ktlint used: 0.50.0
* Kotlin version = 1.9.0
* Gradle version = 8.1.1

**Comment by paul-dingemans** [comment]
> This snippet represents good formatting according to IntelliJ IDEA and should survive after ktlint

This assumption does not hold true for ktlint. Ktlint is more strict than IntelliJ IDEA and disapproves the usage of the elvis operator as the last code element on a line.

The correct assumption is that code produced by Ktlint for code styles `intellij_idea` and `android_studio` should not conflict with the IntelliJ IDEA default formatter. Note that this assumption is also dropped for the code style `ktlint_official` as the the IntelliJ IDEA default code style has formatting errors which after years are still not fixed.
